src/components/PostsList/PostsList.tsx

```diff
@@ -0,0 +1,89 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+
+type Props = {
+  posts: Post[],
+  selectedPost: Post | null,
+  setSelectedPost: (post: Post | null) => void,
+  setIsErrorComments: (v: boolean) => void,
+  setIsFormShown: (v: boolean) => void,
+  setComments: (post: Comment[]) => void,
+};
+
+export const PostsList: React.FC<Props> = ({
+  posts,
+  selectedPost,
+  setSelectedPost,
+  setIsErrorComments,
+  setIsFormShown,
+  setComments,
+}) => {
+  const handleSelectPost = (post: Post) => {
+    if (selectedPost?.id !== post.id) {
+      setSelectedPost(post);
+      setIsFormShown(false);
+
+      return;
+    }
+
+    setIsErrorComments(false);
+    setSelectedPost(null);
+    setComments([]);
+  };
+
+  const handleButtonTitle = (id: number) => {
```

Comment:
```suggestion
  const getButtonTitle = (id: number) => {
```
there you return the title, so the correct prefix will be `get`. You can handle events, etc